### PR TITLE
fix: Update user guide based on changes to UI

### LINKED ISSUE #2519: Update user guide based on changes to UI
Mainly due to migration from Angular to React

### PR DESCRIPTION
**Pull request by jpav**
Fixes #2519

### CODE REVIEW

**Comment by TovaCohen** [comment]
Hi @jpav - I haven't worked on AtlasMap and Syndesis doc for a while. Melissa Flinn is the writer for Syndesis/Fuse Online now. However, are these updates just for the AtlasMap user guide - or will they need to be carried over to the Syndesis content? If they are just for AtlasMap, I would happy to review this. But if they will need to go to Syndesis doc, then Melissa is the doc person you want to review this.  Also, the diffs show changes to every file because of spaces at the end of lines, so it is hard to find what the actual changes are. Any chance that you can do something so we can see the actual changes easily?

**Comment by jpav** [comment]
Hi, @TovaCohen, my understanding is this is specifically for AtlasMap standalone, so I wouldn't think it was referenced in any way by Syndesis. Regarding Melissa, I don't see an ID for her to use to mark her as a reviewer. And finally, I failed to notice the ending spaces; I will definitely fix that.

**Comment by gashcrumb** [comment]
I've gone ahead and added @MelissaFlynn to the atlasmap org, it'd be good to keep her in the loop on doc changes for sure.

**Comment by jpav** [comment]
I'd actually like to remove the trailing whitespace going forward, but I did find in the 'gear' settings icon under the 'File Changed' tab, you can select to ignore all whitespace in the file comparisons. Let me know if that doesn't work for you.

File: docs/src/main/asciidoc/user-guide/topics/applying-conditions-to-mappings.adoc [hunk 1]
```diff
@@ -1,193 +1,177 @@
...
-. Edit the expression input field to specify the conditional 
-expression that you want the data mapper to apply to the 
-mapping. Details about supported conditional expressions follow 
+. Edit the expression input field to specify the conditional
+expression that you want the data mapper to apply to the
+mapping. Details about supported conditional expressions follow
 this procedure.
 +
-As you specify the expression, you can: 
+As you specify the expression, you cane nter `@` and start to enter the name of a field.
```
Comment (pleacu) [inline docs/src/main/asciidoc/user-guide/topics/applying-conditions-to-mappings.adoc:66]: typo: you cane nter

File: docs/src/main/asciidoc/user-guide/topics/applying-conditions-to-mappings.adoc [hunk 2]
```diff
@@ -1,193 +1,177 @@
...
 
-. Enter sample data in the source preview input field(s) 
+. Enter sample data in the source preview input field(s)
 to ensure that the target field or target fields
-get(s) the correct value. 
+get(s) the correct value.
 
 . Optionally, apply transformations to one or more source or target fields
 that are in the mapping:
-.. In the *Mapping Details* panel, find the field that you want to apply
```
Comment (pleacu) [inline docs/src/main/asciidoc/user-guide/topics/applying-conditions-to-mappings.adoc:120]: In the Mapping Details panel, find the field that you want to apply
- Select the bolt icon and select the transformations from the pull-down menu.
- ...

File: docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc [hunk 3]
```diff
@@ -4,21 +4,15 @@
 After all mappings are defined, save the mappings by exporting
 them to an AtlasMap `.adm` catalog file:
 
-. In the AtlasMap top tool bar, click 
-image:ExportIcon.png[Export]. 
+. Click the *AtlasMap* menu and *Export the current mappings and support files into a catalog (.adm)*
 +
-This captures your workspace in a `.adm` catalog file. 
+image:AtlasMapMenu.png[AtlasMap menu].
++
```
Comment (pleacu) [inline docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc:10]: You can't select a folder to put the .adm file into.  Web-based apps shouldn't wrote into user-defined folders.  It always writes the adm file into the Downloads folder.

File: docs/src/main/asciidoc/user-guide/topics/quickstart.adoc [hunk 4]
```diff
@@ -24,45 +24,49 @@ following format:
...
+.. At the top of the *Target* panel, click
+image:Import-Data-Shape.png[Import]
+to import a different JSON or XML file that you can map fields to.
+.. To map fields:
+... Navigate to a source field and click image:CreateNewMapping.png[Create new napping]
++
+The *Mapping Details* panel will slide in from the right.
++
+image:MappingDetails.png[Mapping Details]
+... Navigate to a target field and click image:ConnectToSelectedMapping.png[Connect to the selected mapping] to add it to the selected mapping.
```
Comment (pleacu) [inline docs/src/main/asciidoc/user-guide/topics/quickstart.adoc:64]: Or simply drag the selected target field back to the originating source field for the mapping.

**Review by pleacu — CHANGES_REQUESTED** [review]
A couple of changes...

File: docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc (hunk 3 above)
Comment (jpav) [reply docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc:10]: I see now that is correct, but I'm confused. Why do we allow the user to select a folder in VSC and not in standalone?

File: docs/src/main/asciidoc/user-guide/topics/quickstart.adoc (hunk 4 above)
Comment (jpav) [reply docs/src/main/asciidoc/user-guide/topics/quickstart.adoc:64]: Ah, didn't know about this ability. I'll have to update this in a few places.

**Comment by pleacu** [comment]
VSCode has the notion of a workspace built into it.  The stand-alone
AtlasMap is a React web app where the idea is not typically not let people
pick where they want to write data.

**Comment by jpav** [comment]
I understand the reasoning, but I would disagree about the 'typically' portion. There are a myriad websites that allow the user to select where to download/export data. I'll submit a separate issue for this and correct the documentation for how things work now.

File: docs/src/main/asciidoc/user-guide/topics/combine-multiple-source-fields-into-one-target-field.adoc [hunk 5]
```diff
@@ -13,50 +13,31 @@ and the separator between parts, such as a space or comma. See
...
-
-. If you mapped a source field to each part of the compound target field,
-then skip to the next step.
-+
-If the target field expects data that is not available to be mapped, then in the
-*Mapping Details* panel, edit the index of each
-source field so that it is the same as the index of the corresponding data
-in the compound target field. The data mapper automatically adds
-padding fields as needed to indicate missing data.
+fields are in the order as expected to appear in the target field.
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/combine-multiple-source-fields-into-one-target-field.adoc:34]:
```suggestion
fields are in the order in which they are expected to appear in the target field.
```

File: docs/src/main/asciidoc/user-guide/topics/combine-multiple-source-fields-into-one-target-field.adoc [hunk 6]
```diff
@@ -13,50 +13,31 @@ and the separator between parts, such as a space or comma. See
...
 +
-If you accidentally create too many padding fields, click the trash
-icon on each extra padding field to delete it.
+If necessary, increment or decrement the index number of a source field to achieve the desired order.
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/combine-multiple-source-fields-into-one-target-field.adoc:36]:
```suggestion
If necessary, increase or decrease the index number of a source field to achieve the desired order.
```

File: docs/src/main/asciidoc/user-guide/topics/descriptions-of-atlasmap-icons.adoc [hunk 7]
```diff
@@ -1,30 +1,26 @@
...
+|image:FolderIcon.png[Folder icon]
 |Complex type that you can expand
 
-|image:Lightning-Icon.png[Lightning Bolt]
+|image:LightningBoltIcon.png[Lightning bolt iicon]
 |There is a transformation that is being applied to this field.
 
 |===
+
+There are also several toolbar buttons that allow you to choose what should be displayed, such as datatypes, sample data previews,
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/descriptions-of-atlasmap-icons.adoc:25]:
```suggestion
There are also several toolbar buttons that allow you to choose what should be displayed, such as data types, sample data previews,
```

File: docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc [hunk 8]
```diff
@@ -4,21 +4,15 @@
 After all mappings are defined, save the mappings by exporting
 them to an AtlasMap `.adm` catalog file:
 
-. In the AtlasMap top tool bar, click 
-image:ExportIcon.png[Export]. 
+. Click the *AtlasMap* menu and *Export the current mappings and support files into a catalog (.adm)*
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc:7]:
```suggestion
. Click the *AtlasMap* menu and select *Export the current mappings and support files into a catalog (.adm)*
```

File: docs/src/main/asciidoc/user-guide/topics/importing-java-archive-files-into-atlasmap.adoc [hunk 9]
```diff
@@ -38,36 +38,36 @@ public class GeoLocation {
...
+For example, invoke the following commands to create the `Bicycle.jar` file:
 +
 ----
 javac -cp io.paul:. -d . GeoLocation.java Bicycle.java
 jar cvf ../Bicycle.jar *
 ----
 
-. In the main tool bar, click 
-image:Import-Data-Shape.png[Import].
+. In the main tool bar, click the *AtlasMap* menu and *Import a catalog (.adm)*.
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/importing-java-archive-files-into-atlasmap.adoc:51]:
```suggestion
. In the main toolbar, click the *AtlasMap* menu and select *Import a catalog (.adm)*.
```

File: docs/src/main/asciidoc/user-guide/topics/quickstart.adoc [hunk 10]
```diff
@@ -24,45 +24,52 @@ following format:
...
++
+image:MappingDetails.png[Mapping Details]
+... Either:
+.... Navigate to a target field and click image:ConnectToSelectedMapping.png[Connect to the selected mapping] to add it to the
+selected mapping, or
+.... Drag and drop a target field on the source field to add the target field to the mapping.
+.. With a data mapping selected, explore the optional transformations.
+.. To preview a data mapping result, click
+image:ShowMappingPreview.png[Show Mapping Preview] and enter sample data in source *Mapping preview* fields.
+.. To save your work, click the *AtlasMap* menu and *Export the current mappings and support files into a catalog (.adm)*
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/quickstart.adoc:71]:
```suggestion
.. To save your work, click the *AtlasMap* menu and select *Export the current mappings and support files into a catalog (.adm)*
```

File: docs/src/main/asciidoc/user-guide/topics/resetting-data-files-in-atlasmap.adoc [hunk 11]
```diff
@@ -1,20 +1,19 @@
...
 image:Reset1.png[Remove the specified file]
 
 To remove all imported files, as well as all data mappings:
 
-. In the AtlasMap top menu bar, click 
-image:EditorSettings.png[Settings].
-. In the popup menu, click *Reset All*. 
-. In the confirmation dialog, click *Reset*. 
+. Click the *AtlasMap* menu.
+. In the popup menu, click *Reset all mappings and clear all imported documents*.
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/resetting-data-files-in-atlasmap.adoc:16]:
```suggestion
. In the pop-up menu, select *Reset all mappings and clear all imported documents*.
```

File: docs/src/main/asciidoc/user-guide/topics/separate-one-source-field-into-multiple-target-fields.adoc [hunk 12]
```diff
@@ -13,43 +13,30 @@ and the separator between parts, such as a space or comma. See
...
-. In the *Sources* panel, click the field whose content you want to separate.
-. In the *Target* panel, click the first field that you want to separate the
-source field data into.
-. In the *Target* panel, for each additional target field that you want
-to contain some of the data from the source field, hover over the field and
-press *CTRL-Mouse1* (*CMD-Mouse1* on MacOS) to select it.
+
+. In the *Source* panel, navigate to the field that contains the content you want to separate and click
+image:CreateNewMapping.png[Create new mapping].
+. In the *Target* panel, for each field you want to have map part of the source field's content into, navigate to the field and
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/separate-one-source-field-into-multiple-target-fields.adoc:19]:
```suggestion
. In the *Target* panel, for each field into which you want to map part of the source field's content, navigate to the field and
```

File: docs/src/main/asciidoc/user-guide/topics/separate-one-source-field-into-multiple-target-fields.adoc [hunk 13]
```diff
@@ -13,43 +13,30 @@ and the separator between parts, such as a space or comma. See
...
-
-. If you mapped each part of the compound source field to a target
-field, then skip to the next step.
-+
-If the source field contains data that you do not need, then in the
-*Mapping Details* panel, edit the index of each
-target field so that it is the same as the index of the corresponding data
-in the compound source field. The data mapper automatically adds
-padding fields as needed to indicate unwanted data.
+If necessary, increment of decrement the indexes next to a target field to achieve the desired order.
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/separate-one-source-field-into-multiple-target-fields.adoc:35]:
```suggestion
If necessary, increase or decrease the indexes next to a target field to achieve the desired order.
```

File: docs/src/main/asciidoc/user-guide/topics/transform-target-data.adoc [hunk 14]
```diff
@@ -9,17 +9,17 @@ letter of a data value is uppercase.
 
 .Procedure
 
-. Map the fields. This can be a one-to-one mapping, a combination mapping, 
+. Map the fields. This can be a one-to-one mapping, a combination mapping,
 or a separation mapping.
 
 . In the *Mapping Details* panel, under *Sources* or under *Targets*,
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/transform-target-data.adoc:15]: I'm wondering if the Mapping Details panel still has "Sources" as a heading or whether it changed to "Source" like the main panel.

**Comment by TovaCohen** [comment]
Hi @jpav - I did a copy edit and submitted a few comments. Overall, the doc looks great! I did not read the unchanged content - but I did have one comment on something that just caught my eye.

File: docs/src/main/asciidoc/user-guide/topics/transform-target-data.adoc (hunk 14 above)
Comment (jpav) [reply docs/src/main/asciidoc/user-guide/topics/transform-target-data.adoc:15]: It did not. It is plural in Mapping Details.

File: docs/src/main/asciidoc/user-guide/topics/applying-conditions-to-mappings.adoc [hunk 15]
```diff
@@ -1,193 +1,178 @@
...
-. Edit the expression input field to specify the conditional 
-expression that you want the data mapper to apply to the 
-mapping. Details about supported conditional expressions follow 
+. Edit the expression input field to specify the conditional
+expression that you want the data mapper to apply to the
+mapping. Details about supported conditional expressions follow
 this procedure.
 +
-As you specify the expression, you can: 
+As you specify the expression, you cane enter `@` and start to enter the name of a field.
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/applying-conditions-to-mappings.adoc:66]: Typo: replace "cane" with "can"

File: docs/src/main/asciidoc/user-guide/topics/separate-one-source-field-into-multiple-target-fields.adoc [hunk 16]
```diff
@@ -13,43 +13,30 @@ and the separator between parts, such as a space or comma. See
...
-
-. If you mapped each part of the compound source field to a target
-field, then skip to the next step.
-+
-If the source field contains data that you do not need, then in the
-*Mapping Details* panel, edit the index of each
-target field so that it is the same as the index of the corresponding data
-in the compound source field. The data mapper automatically adds
-padding fields as needed to indicate unwanted data.
+If necessary, increase of decrease the indexes next to a target field to achieve the desired order.
```
Comment (TovaCohen) [inline docs/src/main/asciidoc/user-guide/topics/separate-one-source-field-into-multiple-target-fields.adoc:35]: Typo:  replace "of" with "or"

**Comment by TovaCohen** [comment]
Hi @jpav, I found two typos. Otherwise, this LGTM!

**Comment by jpav** [comment]
Thanks, @TovaCohen. I've corrected those errors. If you could approve the PR, I'll get it checked in.

**Review by TovaCohen — APPROVED** [review]
LGTM!

File: docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc [hunk 17]
```diff
@@ -4,21 +4,15 @@
...
+image:AtlasMapMenu.png[AtlasMap menu].
 +
-This captures your workspace in a `.adm` catalog file. 
-
 . In the *Export mappings* field, accept the default name or enter
 a name for the exported `.adm` file. The default name is `atlasmap-mapping.adm`.
 
 . Click *OK*.
++
+The catalog will be saved in the user's *Downloads* folder.
```
Comment (MelissaFlinn) [inline docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc:16]: change "the user's" to "your"

File: docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc [hunk 18]
```diff
@@ -4,21 +4,15 @@
...
 
-AtlasMap places the exported file in the `Downloads` directory. 
-
-[TIP]
-To set AtlasMap to the state at which you exported data mappings, 
-import a `.adm` catalog file by clicking 
-image:Import-Data-Shape.png[Import]
-in the AtlasMap top tool bar. Navigate to the `.adm` file 
-and click *Open*. 
+. Note, you can later import the .adm file via the *AtlasMap* menu.
```
Comment (MelissaFlinn) [inline docs/src/main/asciidoc/user-guide/topics/exporting-mappings-to-an-atlasmap-catalog-file.adoc:18]: change "via" to "by using"

File: docs/src/main/asciidoc/user-guide/topics/quickstart.adoc [hunk 19]
```diff
@@ -24,45 +24,52 @@ following format:
...
-This file contains the mapping definitions. 
+. Experiment with the data mapper:
+.. At the top of the *Source* panel, click
+image:Import-Data-Shape.png[Import]
+to import a JSON or XML file that you can map fields from.
+.. At the top of the *Target* panel, click
+image:Import-Data-Shape.png[Import]
+to import a different JSON or XML file that you can map fields to.
+.. To map fields:
+... Navigate to a source field and click image:CreateNewMapping.png[Create new napping], or
```
Comment (MelissaFlinn) [inline docs/src/main/asciidoc/user-guide/topics/quickstart.adoc:59]: do you need to delete the ",or" at the end of this line?

**Review by MelissaFlinn — COMMENTED** [review]
Sorry it has taken me so long to reply. 
Like Tova, I scanned through the changed text only. 
I added a few comments about minor things that could be changed. 
I agree - your tip about hiding the white spaces was very helpful. :)

**Comment by jpav** [comment]
Thanks, @MelissaFlinn, those were good catches. Your turn to approve changes :-).

**Review by TovaCohen — APPROVED** [review]
LGTM again :-)

**Comment by jpav** [comment]
@pleacu, Looks like this is waiting for your approval now.

**Review by MelissaFlinn — APPROVED** [review]
Sorry it has taken me so long to reply. 
Like Tova, I scanned through the changed text only. 
I added a few comments about minor things that could be changed. 
I agree - your tip about hiding the white spaces was very helpful. :)

**Review by pleacu — APPROVED** [review]
+1 - tech doc approves.

**Pull request merged**